zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -14,193 +14,240 @@
+        # Verify that the operation amount matches merchFunding. 
+        sp.verify(sp.amount == self.data.merchBal)
+        # Set the channel status to OPEN.
+        self.data.status = OPEN
 
-    # reclaimFunding allows the customer or merchant to withdraw funds
-    # if the other party has not funded their side of the channel yet.
+    # reclaimFunding allows the customer to withdraw their funds
+    # if the merchant has not funded their side of the channel yet.
     @sp.entry_point
     def reclaimFunding(self):
-        sp.verify(self.data.status == AWAITING_FUNDING)
-        sp.if self.data.custAddr == sp.sender:
-            sp.verify(self.data.custBal == self.data.custFunding)
-            sp.send(self.data.custAddr, self.data.custBal)
-            self.data.custBal = sp.tez(0)
-        sp.if self.data.merchAddr == sp.sender:
-            sp.verify(self.data.merchBal == self.data.merchFunding)
-            sp.send(self.data.merchAddr, self.data.merchBal)
-            self.data.merchBal = sp.tez(0)
+        # Only allow the customer to call the entrypoint.
+        sp.verify(self.data.custAddr == sp.sender)
+        # Verify that the channel status is AWAITING_MERCH_FUNDING. For the contract to have a 
+        # status of AWAITING_MERCH_FUNDING, the following must be true:
+        #     - it is a dual funded channel,
+        #     - customer must have already funded their side of the channel, 
+        #     - the merchant has not funded their side.
+        sp.verify(self.data.status == AWAITING_MERCH_FUNDING)
+        # Send the customer’s balance back to custAddr.
+        sp.send(self.data.custAddr, self.data.custBal)
+        # Set the channel status to FUNDING_RECLAIMED.
+        self.data.status = FUNDING_RECLAIMED
  
     # expiry can be called by the merchant to initiate channel closure.
     # The customer should call custClose using the latest state. Otherwise,
     # after the delay expires, the merchant will be able to claim all the
     # funds in the channel using merchClaim.
     @sp.entry_point
     def expiry(self):
+        # Only allow the merchant to call the entrypoint.
         sp.verify(self.data.merchAddr == sp.sender)
+        # Verify that the contract status is OPEN.
         sp.verify(self.data.status == OPEN)
+        # Record the time when the delay period will expire with self.data.delayExpiry.
+        # The time is calculated by adding the number of seconds in selfDelay to timestamp of
+        # the block it is confirmed in.
         self.data.delayExpiry = sp.now.add_seconds(self.data.selfDelay)
         self.data.status = EXPIRY
  
-    # merchClaim can be called by the merchant if the customer has not called
-    # custClose before the delay period has expired.
+    # merchClaim can only be called by the merchant after the expiry entrypoint has been called and # the delay period defined by selfDelay has elapsed. If the customer calls custClose, merchClaim
```

Comment:
Yes. I expanded the comment a bit. In general when we talk about an entrypoint being called, I think it's implied that the operation was confirmed on chain. Otherwise, if it wasn't confirmed, then that part of the contract would not have been executed and it wouldn't have been 'called'. 
```
    # merchClaim can only be called by the merchant after the expiry entrypoint has been called and # the delay period defined by selfDelay has elapsed. If merchClaim is called, the total 
    # channel balance is disbursed to the merchant. The customer may call custClose during the 
    # delay period. If custClose is called before merchClaim, merchClaim can no longer be called.
```

Reply:
OK. I think we've been using "call" to also mean "create the transaction that calls the entrypoint".... which is not necessarily successful.